omod/src/main/webapp/resources/js/view/operation.js

```diff
@@ -178,27 +188,33 @@ define(
 
             updateStatus: function(status) {
                 var self = this;
-
                 // Post the status change using the raw ajax request. This just sends up the changed property, status.
-                $.ajax({
-                    type: 'POST',
-                    url: this.model.url(),
-                    data: '{"status":"' + status + '"}',
-                    success: function(data) {
-                        // Fetch the updated model
-                        self.model.fetch({
-                            success: function() {
-                                // Once the fetch is complete, sync the changes back to the list and close this edit view
-                                self.model.trigger("sync");
-
-                                self.cancel();
-                            }
-                        });
-                    },
-                    error: function(model, resp) { openhmis.error(model, resp); },
-                    contentType: "application/json",
-                    dataType: 'json'
-                });
+               var cancelReason;
+                if (this.model.cancelReason) {
+                    cancelReason = this.model.cancelReason;
```

Comment:
Instead of just setting a cancelReason variable, you could create a data variable and put the status and, if needed, the cancelReason in it.